plugins/database/src/db_plugin.cpp

```diff
@@ -14762,6 +14766,181 @@ auto db_update_ticket_write_byte_count_op(
+            irods::log(LOG_DEBUG9, fmt::format("before:[{}]", before.dump()));
+            irods::log(LOG_DEBUG9, fmt::format("after:[{}]", after.dump()));
+
+            // We have to capture the strings first so that binding to the cllBindVars does not
+            // rely on temporary variables.
+            data_repl_num = after.at("data_repl_num").get<std::string>();
+            data_version = after.at("data_version").get<std::string>();
+            data_type_name = after.at("data_type_name").get<std::string>();
+            data_size = after.at("data_size").get<std::string>();
+            data_path = after.at("data_path").get<std::string>();
+            data_owner_name = after.at("data_owner_name").get<std::string>();
+            data_owner_zone = after.at("data_owner_zone").get<std::string>();
+            data_is_dirty = after.at("data_is_dirty").get<std::string>();
+            data_status = after.at("data_status").get<std::string>();
+            data_checksum = after.at("data_checksum").get<std::string>();
+            data_expiry_ts = after.at("data_expiry_ts").get<std::string>();
+            data_map_id = after.at("data_map_id").get<std::string>();
+            data_mode = after.at("data_mode").get<std::string>();
+            r_comment = after.at("r_comment").get<std::string>();
+            create_ts = after.at("create_ts").get<std::string>();
+            modify_ts = after.at("modify_ts").get<std::string>();
+            resc_id = after.at("resc_id").get<std::string>();
+
+            where_data_id = before.at("data_id").get<std::string>();
+            where_resc_id = before.at("resc_id").get<std::string>();
+
+            // Assign bind variables for execution shortly...
+            cllBindVars[cllBindVarCount++] = data_repl_num.data();
+            cllBindVars[cllBindVarCount++] = data_version.data();
+            cllBindVars[cllBindVarCount++] = data_type_name.data();
+            cllBindVars[cllBindVarCount++] = data_size.data();
+            cllBindVars[cllBindVarCount++] = data_path.data();
+            cllBindVars[cllBindVarCount++] = data_owner_name.data();
+            cllBindVars[cllBindVarCount++] = data_owner_zone.data();
+            cllBindVars[cllBindVarCount++] = data_is_dirty.data();
+            cllBindVars[cllBindVarCount++] = data_status.data();
+            cllBindVars[cllBindVarCount++] = data_checksum.data();
+            cllBindVars[cllBindVarCount++] = data_expiry_ts.data();
+            cllBindVars[cllBindVarCount++] = data_map_id.data();
+            cllBindVars[cllBindVarCount++] = data_mode.data();
+            cllBindVars[cllBindVarCount++] = r_comment.data();
+            cllBindVars[cllBindVarCount++] = create_ts.data();
+            cllBindVars[cllBindVarCount++] = modify_ts.data();
+            cllBindVars[cllBindVarCount++] = resc_id.data();
+
+            cllBindVars[cllBindVarCount++] = where_data_id.data();
+            cllBindVars[cllBindVarCount++] = where_resc_id.data();
```

Comment:
Do we bind strings for all database operations?

I'm curious to know if many/all of the existing iRODS DB functions do this. If I had to guess, we probably bind strings for every DB op due to the use of `cllBindVars`.

If all supported databases accept strings for every column, it may be worth changing the nanodbc-based API plugins to bind everything as a string as well. We'd likely pay the cost of conversion (i.e. string -> db column type) by the DB, but we'd get the benefit of not having to deal with DB differences. Plus, the code would be a little simpler.

Reply:
Yes, `cllBindVars` is an array of `const char*`. I'm pretty sure everything that calls `cmlExecuteNoAnswer` uses `cllBindVars`, but I based this implementation mainly off of `db_mod_data_obj_meta_op`.

Reply:
If it works for nanodbc then that seems fine to me. Wouldn't be any worse than the current implementation from my view.